shared/kubernetes/provider.go

```diff
@@ -419,6 +484,49 @@ func (provider *Provider) CreateMizuRBACNamespaceRestricted(ctx context.Context,
 	return nil
 }
 
+func (provider *Provider) CreateDaemonsetPatchRBAC(ctx context.Context, namespace string, serviceAccountName string, roleName string, roleBindingName string, version string) error {
```

Comment:
Why is this called `Patch`? It allows other verbs as well.
Maybe `DaemonsetRBAC` would be more appropriate?